pkg/segment/segexecution.go

```diff
@@ -459,11 +461,54 @@ func ExecuteAsyncQuery(root *structs.ASTNode, aggs *structs.QueryAggregators, qi
 	}
 
 	go func() {
-		_ = executeQueryInternal(root, aggs, qid, qc, rQuery)
+		if config.IsNewQueryPipelineEnabled() {
+			_ = executeNewPipelineQueryInternal(root, aggs, qid, qc)
+		} else {
+			_ = executeQueryInternal(root, aggs, qid, qc, rQuery)
+		}
 	}()
 	return rQuery.StateChan, nil
 }
 
+func executeNewPipelineQueryInternal(root *structs.ASTNode, aggs *structs.QueryAggregators, qid uint64, qc *structs.QueryContext) *structs.NodeResult {
```

Comment:
same comment as above, rename this func based on what name you give for the var above

Reply:
addressed in the latest commit.